src/encoded/static/components/static-pages/components/StatisticsPageViewBody.js

```diff
@@ -0,0 +1,1092 @@
+        if (!refresh && externalTermMap && _.keys(externalTermMap).length > 0) return;
+
+        ajax.load('/search/?type=Award&limit=all', (resp)=>{
+            this.setState({
+                'externalTermMap' : _.object(_.map(resp['@graph'] || [], function(award){
+                    return [ award.center_title, award.project !== '4DN' ];
+                }))
+            });
+        });
+    }
+
+    render(){ return <StatsViewController {...this.props} {...this.state} />; }
+
+}
+
+
+
+class UsageChartsCountByDropdown extends React.PureComponent {
+
+    constructor(props){
+        super(props);
+        this.handleSelection = this.handleSelection.bind(this);
+    }
+
+    handleSelection(evtKey, evt){
+        const { changeCountByForChart, chartID } = this.props;
+        changeCountByForChart(chartID, evtKey);
+    }
+
+    render(){
+        const { countBy, chartID } = this.props;
+        const currCountBy = countBy[chartID];
+
+        const menuOptions = new Map();
+
+        if (chartID === 'experiment_set_views' || chartID === 'file_views'){
+            menuOptions.set('views',        <React.Fragment><i className="icon fas icon-fw icon-eye mr-1"/>Detail View</React.Fragment>);
+            menuOptions.set('list_views',   <React.Fragment><i className="icon fas icon-fw icon-list mr-1"/>Appearance within first 25 Search Results</React.Fragment>);
+            menuOptions.set('clicks',       <React.Fragment><i className="icon far icon-fw icon-hand-point-up mr-1"/>Search Result Click</React.Fragment>);
+        } else if (chartID === 'file_downloads'){
+            menuOptions.set('experiment_type', <React.Fragment><i className="icon far icon-fw icon-folder mr-1"/>Experiment Type</React.Fragment>);
+            menuOptions.set('geo_country',     <React.Fragment><i className="icon fas icon-fw icon-globe mr-1"/>Country</React.Fragment>);
+            menuOptions.set('filetype',     <React.Fragment><i className="icon far icon-fw icon-file-alt mr-1"/>File Type</React.Fragment>);
+        } else {
+            menuOptions.set('views',    <React.Fragment><i className="icon icon-fw fas icon-eye mr-1"/>View</React.Fragment>);
+            menuOptions.set('sessions', <React.Fragment><i className="icon icon-fw fas icon-user mr-1"/>User Session</React.Fragment>);
+        }
+
+        const dropdownTitle = menuOptions.get(currCountBy);
+
+        return (
+            <div className="inline-block mr-05">
+                <DropdownButton data-tip="Count By" size="sm" id={"select_count_for_" + chartID}
+                    onSelect={this.handleSelection} title={dropdownTitle}>
+                    {_.map([ ...menuOptions.entries() ], function([ k, title ]){
+                        return <DropdownItem eventKey={k} key={k}>{ title }</DropdownItem>;
+                    })}
+                </DropdownButton>
+            </div>
+        );
+    }
+}
+
+
+export const UsageStatsView = React.memo(function UsageStatsView(props){
```

Comment:
Is this view particularly slow, hence why `React.memo` is used here? Or is it a stylistic decision? Seems like there is a lot going on here.

Reply:
It's not particularly slow no. Might not be best place to have used React.memo.
If had time, ideally we'd refactor this to get rid of `aggregationsToChartData` and put that logic into individual charts like "FileDownloadsChart" , and then we could memoize "FileDownloadsChart" (and/or logic within it, such as the aggregation) and that would make a very noticeable improvement in simplicity+clarity of this logic as a whole and make better/good use of memoization. Alas the priority of this is.. not high..

Reply:
Well a fun exercise to do - maybe a useful starter feature for me at some point when we have more time.